packages/unmock-core/src/__tests__/generator.test.ts

```diff
@@ -1,140 +1,164 @@
+});
 
-    resp = createResponse({
-      host: "cloud.filestackapi.com",
-      method: "get",
-      path: "/prefetch",
-      pathname: "/prefetch",
-      protocol: "https",
-      query: {},
-    });
-    expect(resp).toBeDefined();
-    if (resp !== undefined) {
-      expect(resp.statusCode).toEqual(200);
-      expect(resp.body).toEqual('"prefetch"');
-    }
-  });
+test("ref name works", () => {
+  expect(refName({ $ref: "#/components/schemas/Foo"})).toBe("Foo");
+});
 
-  it("Sets a state with a function and generates accordingly", () => {
-    const createResponse = responseCreatorFactory({
-      options: mockOptions,
-      store: backend.serviceStore,
-    });
-    services.petstore.state({ id: () => "foo", $size: 5, $code: 200 });
-    let resp = createResponse({
-      host: "petstore.swagger.io",
-      method: "get",
-      path: "/v1/pets",
-      pathname: "/v1/pets",
-      protocol: "http",
-      query: {},
-    });
-    expect(resp).toBeDefined();
-    if (resp && resp.body !== undefined) {
-      JSON.parse(resp.body).forEach((pet: any) =>
-        expect(pet.id).toEqual("foo"),
-      );
-    } else {
-      throw new Error("Response body was undefined?");
-    }
+test("key lens works", () => {
+  expect(keyLens().get([1, 2])).toBe(1);
+});
+
+test("operation optional works", () => {
+  expect(operationOptional.getOption({
+    get: { responses: { 100: { description: "hello"} }},
+    description: "foo",
+  })).toEqual(some(["get", { responses: { 100: { description: "hello"} }}]));
+  expect(operationOptional.getOption({
+    description: "foo",
+  })).toEqual(none);
+});
+
+const baseO: OpenAPIObject = {
+  openapi: "hello",
+  info: { title: "", version: ""},
+  servers: [{url: "https://hello.api.com"}],
+  paths: {},
+};
+
+test("use if header works", () => {
+  expect(useIfHeader(baseO, { name: "foo", in: "query"})).toEqual(none);
+  expect(useIfHeader(baseO, { name: "foo", in: "header"})).toEqual(some(["foo", {type: "string"}]));
+  expect(useIfHeader(baseO, { name: "foo", in: "header", schema: { type: "number" }}))
+    .toEqual(some(["foo", {type: "number"}]));
+  expect(useIfHeader({
+    ...baseO,
+    components: { schemas: { Foo: { type: "boolean" }}},
+  }, { name: "foo", in: "header", schema: { $ref: "#/components/schemas/Foo" }}))
+    .toEqual(some(["foo", {type: "boolean"}]));
+});
+
+test("identity getter works", () => {
+  expect(identityGetter().get(1)).toBe(1);
+});
 
-    services.petstore.state({ id: () => 1 });
-    resp = createResponse({
-      host: "petstore.swagger.io",
-      method: "get",
-      path: "/v1/pets",
-      pathname: "/v1/pets",
-      protocol: "http",
-      query: {},
-    });
-    expect(resp).toBeDefined();
-    if (resp && resp.body !== undefined) {
-      JSON.parse(resp.body).forEach((pet: any) => expect(pet.id).toEqual(1));
-    } else {
-      throw new Error("Response body was undefined?");
-    }
+test("hoist transformer works", () => {
```

Comment:
So many of the "stuff works" tests here 😬  I think it would also act as documentation if the tests were more like
```js
describe("hoist transformer", () => {
  it("should hoist correctly when matching host", () => { ... });
  it("should result in no-op when not matching host, () => { ... });
});
```